### PR TITLE
[F] Return a shuffled array (#988)

### PR DESCRIPTION
**Pull request by DrWilliamsSteven**
Closes repositive/report-issues#988

Backend for repositive/report-issues#940

### CODE REVIEW

File: src/main/routes/trending-datasets.ts [hunk 1]
```diff
@@ -32,21 +35,16 @@ async function getTrending(datasets: any, refreshTime: number, force: boolean =
...
-
     const { datasets, config } = req.custom_state();
 
     if (!running) {
         running = setInterval(() => getTrending(datasets, config.refreshTime, true), config.refreshTime);
     }
 
-    const result = getTrending(datasets, config.refreshTime)
-
+    const result = getTrending(datasets, config.refreshTime, false)
```
Comment (DrWilliamsSteven) [inline src/main/routes/trending-datasets.ts]: This doesn't need to be false as it is default..

File: src/main/routes/trending-datasets.ts [hunk 2]
```diff
@@ -32,21 +35,16 @@ async function getTrending(datasets: any, refreshTime: number, force: boolean =
...
 
-
-
 export async function handler(req: any, rep: any) {
-
     const { datasets, config } = req.custom_state();
 
     if (!running) {
         running = setInterval(() => getTrending(datasets, config.refreshTime, true), config.refreshTime);
     }
```
Comment (DrWilliamsSteven) [inline src/main/routes/trending-datasets.ts]: At the moment the initial call still returns an empty array. Should this actually be:
```js
if (!running) {
    running = true;
    setInterval(() => getTrending(datasets, config.refreshTime, true), config.refreshTime);
  }
```

Comment (Istar-Eldritch) [inline src/main/routes/trending-datasets.ts]: The initial call will still return an empty array. Running gets initialized as an interval object. We should change it and remove the logic from the handler. I'll show you how we can do it.

File: src/main/utils/shuffleArray.ts [hunk 3]
```diff
@@ -0,0 +1,13 @@
+/**
+ * Function to randomize array element order in-place.
+ * Using Durstenfeld shuffle algorithm.
+ */
+export default function shuffleArray(array: any[]) {
```
Comment (Istar-Eldritch) [inline src/main/utils/shuffleArray.ts]: Can we write unit tests for this function?

**Comment by Istar-Eldritch** [comment]
Civerage fails. I am aproving this one. Well fix the coverage next week

**Pull request merged**